x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedConfigAutoUpdater.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.datafeed;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.Version;
+import org.elasticsearch.action.support.PlainActionFuture;
+import org.elasticsearch.xpack.core.ml.datafeed.DatafeedConfig;
+import org.elasticsearch.xpack.core.ml.datafeed.DatafeedUpdate;
+import org.elasticsearch.xpack.ml.MlAutoUpdateService;
+import org.elasticsearch.xpack.ml.datafeed.persistence.DatafeedConfigProvider;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DatafeedConfigAutoUpdater implements MlAutoUpdateService.UpdateAction {
+
+    private static final Logger logger = LogManager.getLogger(DatafeedConfigAutoUpdater.class);
+    private final DatafeedConfigProvider provider;
+
+    public DatafeedConfigAutoUpdater(DatafeedConfigProvider provider) {
+        this.provider = provider;
+    }
+
+    @Override
+    public boolean minNodeSupported(Version minNodeVersion) {
+        return minNodeVersion.onOrAfter(Version.V_8_0_0);
+    }
+
+    @Override
+    public String getName() {
+        return "datafeed_aggs_updater";
+    }
+
+    @Override
+    public void runUpdate() {
+        PlainActionFuture<List<DatafeedConfig.Builder>> getdatafeeds = PlainActionFuture.newFuture();
+        provider.expandDatafeedConfigs("_all", true, getdatafeeds);
+        List<DatafeedConfig.Builder> datafeedConfigBuilders = getdatafeeds.actionGet();
+        List<DatafeedUpdate> updates = datafeedConfigBuilders.stream()
+            .map(DatafeedConfig.Builder::build)
+            .filter(DatafeedConfig::aggsRewritten)
+            .map(datafeedConfig -> new DatafeedUpdate.Builder()
+                .setAggregations(datafeedConfig.getAggProvider())
+                .setId(datafeedConfig.getId())
+                .build())
+            .collect(Collectors.toList());
+        if (updates.isEmpty()) {
```

Comment:
If there are no datafeed configs, that also means there won't be any updates. 

This is ok, as all new datafeed configs enforce the appropriate date_histogram agg format.